### PR TITLE
Refactor, 100% TestCoverage, add ajv.plugins Option

### LINKED ISSUE #42: Unknown format "float" is used in schema at path "..."
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure it has not already been reported

### Fastify version

3.17.0

### Plugin version

0.1.0

### Node.js version

16.2.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

10.15.7

### Description

Hi, I'm trying to use this plugin to validate the responses generated by my Fastify app. In addition to Fastify, in order to generate programmatically all the routes from an openapi specification, I'm using fastify-openapi-glue.

Each time I try to run my app I get the following error:
`unknown format "float" is used in schema at path "..."` 

In order to solve this issue i tried to use others libraries in order to integrate the missing format, for example:
- ajv-formats
- ajv-openapi

I tryed also to override the instance of ajv used by Fastify with another one.

### Steps to Reproduce

1. Define a openApi specification that uses the float format for one of its properties;
2. Create a Fastify app that uses the previous openApi specification to generate all its routes through the use of fastify-openapi-glue plugin;
3. Inside the app code and before the use of fastify-openapi-glue register the fastify-response-validation plugin to enable the response validation;
4. Run the app and see if the error occours.

### Expected Behavior

Shouldn't throw the error reported in the description.

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #42 

I dont know how to add typings for the onRoute option.

Any suggestions?

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
Resolves #42

**Review by mcollina — APPROVED** [review]
lgtm

File: types/index.test-d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+app.register(plugin, { responseValidation: false });
+app.register(plugin, { ajvPlugins: [ajvFormats] })
+app.register(plugin, { ajvPlugins: [ajvErrors] })
+app.register(plugin, { ajvPlugins: [[ajvErrors, {}]] })
+app.register(plugin, { ajvPlugins: [ajvFormats, [ajvErrors, {}]] })
+
+app.route({
+  method: 'GET',
+  url: '/',
+  responseValidation: false,
```
Comment (Uzlopak) [inline types/index.test-d.ts:22]: this will error, because I dont know how to do the onRoute options  properly.

**Comment by Uzlopak** [comment]
@mcollina 

Can you please assign somebody who knows how to Implement the on Route typings?

**Comment by Uzlopak** [comment]
@climba03003 gave me some hints. Will adapt them when i am back at a Computer.

**Comment by Uzlopak** [comment]
Fixed.

I actually would have preferred to have the same syntax as fastify does. Which would have been 

```
type Options = {
   ajv: {
      customOptions: AjvOptions;
      plugins: (Function | [Function, unknown])[]
   }
}
```

But this should be fine also...

**Comment by Uzlopak** [comment]
If you are willing to make a mayor version release then i would like to put the ajvOptions into the curtomOptions key, to make it 100% identical to the request validation.

**Review by climba03003 — APPROVED** [review]
LGTM.

**Comment by climba03003** [comment]
> If you are willing to make a major version release then i would like to put the ajvOptions into the customOptions key, to make it 100% identical to the request validation.

We should land the refactor first. Then go on for the options consistency in `fastify` and `plugin`.
I love to see the `ajv` options is consistence across `fastify` ecosystem.

**Review by Eomm — APPROVED** [review]
> I love to see the ajv options is consistence across fastify ecosystem.

This plugin could benefit the new feature `request.validateInput` from https://github.com/fastify/fastify/pull/3970

**Comment by Uzlopak** [comment]
Should we integrate this into fastify core?

**Comment by Eomm** [comment]
> Should we integrate this into fastify core on the long run?

No, I don't think so.
I must say that validating the response is fine only in very rare cases in my opinion and should not be the default

**Pull request merged**